### PR TITLE
Feat/minio reader noverify

### PR DESCRIPTION
**Pull request by ferdinandosimonetti**
# Description

The fix allows reading documents from a MinIO instance secured with TLS **even if the instance's certificate isn't trusted**.
It introduces `minio_cert_check` option (defaulting to False, so running it against a self-signed MinIO will be fine).

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/readers/llama-index-readers-minio/llama_index/readers/minio/minio_client/base.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def __init__(
         file_metadata: Optional[Callable[[str], Dict]] = None,
         minio_endpoint: Optional[str] = None,
         minio_secure: bool = False,
+        minio_cert_check: bool = False,
```
Comment (sansmoraxz) [inline llama-index-integrations/readers/llama-index-readers-minio/llama_index/readers/minio/minio_client/base.py:32]: Shouldn't this be `True` by default as a security best practice?

Comment (ferdinandosimonetti) [reply llama-index-integrations/readers/llama-index-readers-minio/llama_index/readers/minio/minio_client/base.py:32]: You're right. I'll fix that ASAP

**Comment by ferdinandosimonetti** [comment]
the same feature was approved as a PR on llamahub about a couple of days ago.

**Review by hatianzhang — APPROVED** [review]
thanks!

**Pull request merged**